src/Foundatio.AWS/Queues/SQSQueue.cs

```diff
@@ -0,0 +1,269 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Amazon;
+using Amazon.Runtime;
+using Amazon.SQS;
+using Amazon.SQS.Model;
+using Foundatio.Logging;
+using Foundatio.Queues;
+using Foundatio.Serializer;
+using Foundatio.Utility;
+using Nito.AsyncEx;
+
+namespace Foundatio.AWS.Queues {
+    public class SQSQueue<T> : QueueBase<T> where T : class {
+
+        private readonly AsyncLock _lock = new AsyncLock();
+
+        private readonly Lazy<AmazonSQSClient> _client;
+        private readonly TimeSpan _workItemTimeout;
+        private readonly TimeSpan _readQueueTimeout;
+        private readonly AWSCredentials _credentials;
+        private readonly RegionEndpoint _regionEndpoint;
+
+        private string _queueUrl;
+
+        private long _enqueuedCount;
+        private long _dequeuedCount;
+        private long _completedCount;
+        private long _abandonedCount;
+        private long _workerErrorCount;
+
+
+        public SQSQueue(
+            string queueName,
+            AWSCredentials credentials = null,
+            RegionEndpoint regionEndpoint = null,
+            TimeSpan? workItemTimeout = null,
+            TimeSpan? readQueueTimeout = null,
+            ISerializer serializer = null,
+            IEnumerable<IQueueBehavior<T>> behaviors = null,
+            ILoggerFactory loggerFactory = null)
+            : base(serializer, behaviors, loggerFactory) {
+
+            _queueName = queueName ?? throw new ArgumentNullException(nameof(queueName));
+
+            _credentials = credentials ?? FallbackCredentialsFactory.GetCredentials();
+            _regionEndpoint = regionEndpoint ?? RegionEndpoint.USEast1;
+            _workItemTimeout = workItemTimeout ?? TimeSpan.FromMinutes(5);
+            _readQueueTimeout = readQueueTimeout ?? TimeSpan.FromSeconds(2);
+
+            _client = new Lazy<AmazonSQSClient>(() => new AmazonSQSClient(_credentials, _regionEndpoint));
+        }
+
+        protected override async Task EnsureQueueCreatedAsync(CancellationToken cancellationToken = new CancellationToken()) {
+            if (!String.IsNullOrEmpty(_queueUrl))
+                return;
+
+            using (await _lock.LockAsync(cancellationToken).ConfigureAwait(false)) {
```

Comment:
That one is not supported, it returns something not supported by the extension method